src/engine/Story.ts

```diff
@@ -1839,10 +1973,14 @@ export class Story extends InkObject {
 
     if (typeof specificVariableName !== "undefined") {
       if (this._variableObservers.has(specificVariableName)) {
-        let observers = this._variableObservers.get(specificVariableName)!;
-
         if (observer !== null) {
-          observers.splice(observers.indexOf(observer), 1);
+          let observers = this._variableObservers.get(specificVariableName)!;
+
+          if (observers !== null) {
+            observers.splice(observers.indexOf(observer), 1);
+          } else {
+            this._variableObservers.delete(specificVariableName);
+          }
```

Comment:
I think we may have a small difference in behavior here? Upstream, after removing the specific observer, if there are no more observers left for that variable, it is removed entirely (l. 2340 +2 in Story.cs).

Reply:
Yeah, you're right something's off here, let me update!